src/Microsoft.AspNetCore.Routing/Constraints/RegexInlineRouteConstraint.cs

```diff
@@ -13,7 +13,7 @@ public class RegexInlineRouteConstraint : RegexRouteConstraint
         /// </summary>
         /// <param name="regexPattern">The regular expression pattern to match.</param>
         public RegexInlineRouteConstraint(string regexPattern)
-            : base(regexPattern)
+            : base("^" + regexPattern + "$")
```

Comment:
Unfortunately, this is a breaking change so we cannot take it.

Perhaps we could reverse the behavior. That is, keep this type unchanged, but have the other one be something like `FullMatchRegexInlineRouteConstraint`.

Thoughts?